### DISCUSSION TITLE
Mod'ing code to work standalone

### DISCUSSION THREAD

**Discussion by becker666** [arendst/Tasmota#10750] [Q&A] [answered]
I see a few users with the same request but have not found a solution.
    - I need to use it at at remote location standalone where there's no internet. I don't care it theh time gets shifted back/forward a few minutes
    - Playing with the current version I see if the internet goes out the local webserver pops every once in a while and the time is reset to 1970 messing up all my timers(which do stay in non-volatile memory)
    - I've been digging thru the source code & what I'm planning on doing is add a check box to that local temp web  with this box it will skip checking NTP , stop the webserver from popping , and call the time command to set it to current time. I can set the timers while setting up as they hold on non-volatile memory. If lose power go back to original code with the4 box unchecked.

Anybody has done this already ??? I don't want to reinvent the wheel.

    Thanks in advance

    Becker

**Comment by s-hadinger** [comment]
Maybe the simplest would be to use `Settings.cfg_timestamp` which olds the time for the last configuration written to Flash. This gives you a first estimate of the date, instead of 1970. However it will mess up your timers after a crash (settings are written at normal reboot).

**Comment by barbudor** [comment]
Can't you have a local NTP server ?
Even without Internet access, may be the Wifi AP can provide a NTP interface
Or you can make a tiny one out of an ESP8266 and a RTC
Even Tasmota can act as a NTP Server when enabling the GPS.

**Comment by meingraham** [comment]
To clarify, you want the time on your device(s) to be close to the actual time or just that they have a time set that's closer to real time than 1970? Are you trying to have something on your device(s) respond to an actual time of day or just to have time/timers available for operation?

**Reply by becker666** [reply]
I just need the sonoffs to turn on/off as set on the timers  which only hold time and days of week I had set timers a couple of months ago and they stayed there on the sonoffs after being unplugged this long.  When internet goes out it resets the epoch & keeps popping the web to setup SSID and stuff every few minutes, anytime it pops that webpage it resets it to 1970 been trying to dig that code from the source code but pretty convoluted once the spot is found sounds easy to add the logic. The RTC chip seems the cheapest/quickest shot as it is only $1 or $2, but  due to pandemic I can try to it all in software???

**Reply by ascillato** [reply]
If you need time and weekdays, you must go for the RTC chip, GPS chip or local NTP server like a Raspberry or your router if it supports it. If not, on every reset or power failure, you will lost the time. You can set the time in tasmota using command `time`, but the time will be lost on every reset or power failure.

**Comment by ascillato** [comment, marked as answer]
Yes, this has been done already. Attach a GPS chip or a RTC chip and you will have all features working. Those are already supported. Time is needed to work properly. Without time, Timers obviously won't work.
